components/contact-list/src/ContactList.svelte

```diff
@@ -130,7 +132,7 @@
   let queryKey: string;
   $: queryKey = JSON.stringify(query);
 
-  $: setHydratedContacts(), contacts, queryKey;
```

Comment:
Hey I believe this will break the ability to toggle between custom and Nylas contacts data 

Use case is like filtering contacts: 
- By default show all your Nylas contacts
- user clicks a filter, so you pass in custom contacts that abide by the filter
- user clears the filter, so you clear your custom contacts and it reverts back to Nylas contacts

I believe removing queryKey from here will break this usecase - the test is failing: https://github.com/nylas/components/runs/4398559856?check_suite_focus=true

this was the orig PR: https://github.com/nylas/components/pull/168

:/ I know there's a lot of reactive statements in here making it kinda messy. @ozsivanov do you have any plans to refactor Contacts?

Reply:
It doesn't actually. I've tested it manually switching between Nylas data and custom. 
The test needs a refactor though. 

Check the loom I go over that. But I haven't got over the filtering.

Reply:
Notice that the query only contains the componentId and an access token.